api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from typing import Type
+import logging
+
+from .. import (
+    inputs,
+    errors,
+    config,
+    ALL_MINIONS
+)
+from ..utils import run_subprocess_cmd
+from ..salt import local_minion_id
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..vendor import attr
+# TODO IMPROVE EOS-8473
+
+from .configure_setup import (
+    SetupType
+)
+from .deploy import (
+    build_deploy_run_args,
+    Deploy
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+deploy_states = dict(
+    prvsnr=[
```

Comment:
In my opinion it would be good to move these to :
`srv/components/system/storage/glusterfs/teardown/init.sls`

Then just make a call to `"system.storage.glusterfs.teardown"` here.

Reply:
We need to unmount from all nodes first then we do delete volumes.

Reply:
splitted into 2 things
1. volume_remove -> unmount and remove volume
2. init.sls -> remove bricks, stop services , remove rpms